File: src/things/Rock.cpp
#include "things/Rock.h"

Rock::Rock(UUID id) : Thing("rock", id)
{
  // ctor
}

Rock::~Rock()
{
  // dtor
}

std::string Rock::_get_description() const
{
  return "small rock";
}

sf::Vector2u Rock::get_tile_sheet_coords(int frame) const
{
  return sf::Vector2u(0, 8);
}

File: resources/things/Rock.lua
-- Definition of special functions for the Rock object type.

require "resources/things/Thing"

Rock = inheritsFrom(Thing, "Rock")
Rock.name = "rock"

Rock.intrinsics.physical_mass = 3

function Rock.get_description()
	return "A stone of any size, especially one small enough to be picked up and used as a projectile."
end